bpf/lbmap/lbmap.go

```diff
@@ -173,13 +208,94 @@ func LookupRevNat(key RevNatKey) (RevNatValue, error) {
 	return revnat.Convert(), nil
 }
 
+func gcd(x, y uint16) uint16 {
+	for y != 0 {
+		x, y = y, x%y
+	}
+	return x
+}
+
+type rrState struct {
+	index int
+	cw    uint16
+}
+
+func next(state *rrState, n int, max uint16, weights []uint16) uint16 {
+	for {
+		state.index = (state.index + 1) % n
+		if state.index == 0 {
+			state.cw = state.cw - 1
+			if state.cw <= 0 {
+				state.cw = max
+			}
+		}
+		if weights[state.index] >= state.cw {
```

Comment:
Well the worst case could be O(gcd of sum of weights * sequence count). So you can assign really bad weights and make the loop count go high. It is the standard algorithm used in many places and I would not be worried about it. The advantage is its an O(1) operation in the data path. Just one lookup. Given the data path only cares about the sequence, we can find out other optimal sequence generation algorithms later. I found few but decided to use this one since its the most common.